csharp/device/Microsoft.Azure.Devices.Client/DeviceClient.cs

```diff
@@ -219,19 +250,34 @@ public AsyncTaskOfMessage ReceiveAsync(TimeSpan timeout)
         /// Deletes a received message from the device queue
         /// </summary>
         /// <returns>The lock identifier for the previously received message</returns>
+
+#if NETMF
```

Comment:
No problem. I just left each method implementation in NETMF near it's counterpart in the full Framework. It's really a matter on how things should be organized.